model/subscriber.go

```diff
@@ -64,3 +70,63 @@ func (s *Subscriber) Copy(dst datastore.Entity) (datastore.Entity, error) {
 func (s *Subscriber) GetCache() datastore.Cache {
 	return nil
 }
+
+// CreateSubscriber creates a subscriber, or returns an error if a subscriber with the
+// given ID already exists.
+func CreateSubscriber(ctx context.Context, store datastore.Store, s *Subscriber) error {
+	key := store.IDKey(typeSubscriber, s.ID)
+	return store.Create(ctx, key, s)
+}
+
+// GetSubscriberByEmail returns the subscriber with the given email if it exists.
+func GetSubscriberByEmail(ctx context.Context, store datastore.Store, email string) (*Subscriber, error) {
+	q := store.NewQuery(typeSubscriber, false)
+	q.FilterField("Email", "=", email)
+	var subs []Subscriber
+	_, err := store.GetAll(ctx, q, &subs)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to get all subscribers: %w", err)
+	}
+
+	if len(subs) == 0 {
+		return nil, datastore.ErrNoSuchEntity
+	}
+
+	if len(subs) > 1 {
+		return nil, errDuplicateSubscriberEmails
+	}
+
+	return &subs[0], err
+}
+
+// GetSubscriber gets the subscriber with the given ID.
+func GetSubscriber(ctx context.Context, store datastore.Store, id int64) (*Subscriber, error) {
+	key := store.IDKey(typeSubscriber, id)
+	sub := &Subscriber{}
+	err := store.Get(ctx, key, sub)
+	return sub, err
+}
+
+// UpdateSubscriber updates the subscriber record with the given subscriber, based on the ID
+// of the passed subscriber.
+func UpdateSubscriber(ctx context.Context, store datastore.Store, subscriber *Subscriber) error {
+	key := store.IDKey(typeSubscriber, subscriber.ID)
+	_, err := store.Put(ctx, key, subscriber)
+	return err
+}
+
+// NewSubscriberID returns a new unique in64 ID which does not already exist in
+// the datastore.
+func NewSubscriberID(ctx context.Context, store datastore.Store) int64 {
```

Comment:
Is there a reason why we do it like this ? We could just use https://pkg.go.dev/github.com/google/uuid ?